kyo-core/shared/src/main/scala/kyo/Adder.scala

```diff
@@ -13,56 +13,75 @@ class LongAdder private[kyo] (private val ref: j.LongAdder) extends AnyVal:
       * @return
       *   Unit
       */
-    inline def add(v: Long)(using Frame): Unit < IO = IO(ref.add(v))
+    inline def add(v: Long)(using inline frame: Frame): Unit < IO = IO(unsafe.add(v))
 
     /** Decrements the sum by one.
       *
       * @return
       *   Unit
       */
-    inline def decrement(using Frame): Unit < IO = IO(ref.decrement())
+    inline def decrement(using inline frame: Frame): Unit < IO = IO(unsafe.decrement())
 
     /** Increments the sum by one.
       *
       * @return
       *   Unit
       */
-    inline def increment(using Frame): Unit < IO = IO(ref.increment())
+    inline def increment(using inline frame: Frame): Unit < IO = IO(unsafe.increment())
 
     /** Returns the current sum.
       *
       * @return
       *   The current sum
       */
-    inline def get(using Frame): Long < IO = IO(ref.sum())
+    inline def get(using inline frame: Frame): Long < IO = IO(unsafe.get())
 
     /** Resets the sum to zero.
       *
       * @return
       *   Unit
       */
-    inline def reset(using Frame): Unit < IO = IO(ref.reset())
+    inline def reset(using inline frame: Frame): Unit < IO = IO(unsafe.reset())
 
     /** Returns the current sum and resets it to zero.
       *
       * @return
       *   The sum before reset,
       */
-    inline def sumThenReset(using Frame): Long < IO = IO(ref.sumThenReset())
+    inline def sumThenReset(using inline frame: Frame): Long < IO = IO(unsafe.sumThenReset())
+
 end LongAdder
 
 object LongAdder:
+
     /** Creates a new LongAdder instance.
       *
       * @return
       *   A new LongAdder
       */
-    def init(using Frame): LongAdder < IO = IO(LongAdder(new j.LongAdder))
+    def init(using frame: Frame): LongAdder < IO = IO(LongAdder(Unsafe.init()))
+
+    opaque type Unsafe = j.LongAdder
```

Comment:
The pattern is offering unsafe APIs like `LongAdder.Unsafe` and `Latch.Unsafe` with methods `unsafe`/`safe` to switch between the regular API and the `Unsafe` version.